synthesizer/src/vm/mod.rs

```diff
@@ -398,7 +398,34 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         self.block_store().pause_atomic_writes()?;
 
         // First, insert the block.
-        self.block_store().insert(block)?;
+        if let Err(insert_error) = self.block_store().insert(block) {
+            if cfg!(feature = "rocks") {
+                // Clear all pending atomic operations so that unpausing the atomic writes
+                // doesn't execute any of the queued storage operations.
+                self.block_store().abort_atomic();
+                // Disable the atomic batch override.
+                // Note: This call is guaranteed to succeed (without error), because `DISCARD_BATCH == true`.
+                self.block_store().unpause_atomic_writes::<true>()?;
+                // Rollback the Merkle tree.
+                self.block_store().remove_last_n_from_tree_only(1).map_err(|removal_error| {
```

Comment:
If there was an issue with adding the block to the block store, why would we need to revert the tree or rollback the block?

Reply:
Oh we might not, I thought there might be partial state that would need rolling back but I'm not yet super familiar with this codepath.

Reply:
added a commit to remove this block